File: readability/readability.py
from math import ceil

numberOfWords = 1
numberOfLetters = 0
numberOfSentences = 0

text = input('Text: ')

for c in text:
    if (c.isalpha()):
        numberOfLetters += 1

    elif c.isspace():
        numberOfWords += 1

    elif c == '.' or c == '?' or c == '!':
        numberOfSentences += 1

    if numberOfWords == 100:
        break

if numberOfWords < 100:
    numberOfLetters = (numberOfLetters / numberOfWords) * 100
    numberOfSentences = (numberOfSentences / numberOfWords) * 100

index = ceil((0.0558 * numberOfLetters) - (0.296 * numberOfSentences) - 15.8)

if index < 1:
    print('Before Grade 1')

elif index > 16:
    print('Grade 16+')

else:
    print(f'Grade {index}')

File: readability/readability.c
#import <stdio.h>
#import <ctype.h>
#import <cs50.h>
#import <string.h>
#import <math.h>

int main(void)
{
    float numberOfWords = 1;
    float numberOfLetters = 0;
    float numberOfSentences = 0;
    int index;

    string text = get_string("Text: ");

    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            numberOfLetters++;
        }

        else if (isspace(text[i]))
        {
            numberOfWords++;
        }

        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            numberOfSentences++;
        }

        if (numberOfWords == 100)
        {
            break;
        }
    }

    if (numberOfWords < 100)
    {
        numberOfLetters = (numberOfLetters / numberOfWords) * 100;
        numberOfSentences = (numberOfSentences / numberOfWords) * 100;
    }

    index = round(0.0588 * numberOfLetters - 0.296 * numberOfSentences - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index > 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", index);
    }

}
